tracer/src/Datadog.Trace/Telemetry/DTOs/ApplicationTelemetryData.cs

```diff
@@ -0,0 +1,28 @@
+﻿// <copyright file="ApplicationTelemetryData.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.Telemetry
+{
+    internal class ApplicationTelemetryData
+    {
+        public string ServiceName { get; set; }
```

Comment:
Since some fields are mandatory while others are not, could we enable nullability on this/other DTO's?

Reply:
![image](https://user-images.githubusercontent.com/18755388/148767339-13f82281-2bf1-4894-9c71-05e0b6744f58.png)

Done in https://github.com/DataDog/dd-trace-dotnet/pull/2241/commits/d18010bd3769c8abd9d86e69fd871f9ddc999e19 though, along with the required constructors and refactoring 😄 Though I suspect that may break a bunch of integration tests that can no longer deserialize the DTOs. We'll see shortly...